### PR TITLE
add mutation and mailing functionalities for share with a friend action

### PR DESCRIPTION
**Pull request by eenwemasor**
added mutation and mailing functionalities for share with a friend action

### CODE REVIEW

File: app/GraphQL/Mutations/ShareWithAFriend.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+use App\Services\ShareWithAFriendService;
+use GraphQL\Type\Definition\ResolveInfo;
+use Nuwave\Lighthouse\Support\Contracts\GraphQLContext;
+
+class ShareWithAFriend
+{
+    /**
+     * @var ShareWithAFriendService
+     */
+    private $share_with_a_friend_service;
```
Comment (omadoyeabraham) [inline app/GraphQL/Mutations/ShareWithAFriend.php]: @enwemasor1995 Change this to camelcase $shareWithAFriendService

File: app/Mail/NewUserRegisteredMail.php [hunk 2]
```diff
@@ -55,6 +55,6 @@ public function __construct(User $user, string $defaultPassword, $registration_s
     public function build()
     {
         return $this->subject('UMC - New User Sign Up')
-            ->markdown('email.signup.user');
+            ->view('email.signup.user');
```
Comment (omadoyeabraham) [inline app/Mail/NewUserRegisteredMail.php]: @enwemasor1995 Why did you change this from markup to view?

File: app/GraphQL/Mutations/ShareWithAFriend.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * Return a value for the field.
+     *
+     * @param  null  $rootValue Usually contains the result returned from the parent field. In this case, it is always `null`.
+     * @param  mixed[]  $args The arguments that were passed into the field.
+     * @param  \Nuwave\Lighthouse\Support\Contracts\GraphQLContext  $context Arbitrary data that is shared between all fields of a single query.
+     * @param  \GraphQL\Type\Definition\ResolveInfo  $resolveInfo Information about the query itself, such as the execution state, the field name, path to the field from the root, and more.
+     * @return mixed
+     */
+    public function  resolve($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
```
Comment (omadoyeabraham) [inline app/GraphQL/Mutations/ShareWithAFriend.php]: Thie function should be called __invoke just like other GraphQL query and mutation classes in this codebase

File: app/Mail/ShareWithAFriendMail.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    /**
+     * Build the message.
+     *
+     * @return $this
+     */
+    public function build()
+    {
+        return $this->subject('Invitation to join UMC Upspring Multipurpose Cooperative')
+            ->markdown('email.share-with-a-friend');
```
Comment (omadoyeabraham) [inline app/Mail/ShareWithAFriendMail.php]: @enwemasor1995 Your view filename should also use camelcase i.e email.shareWithAFriend

File: app/Services/ShareWithAFriendService.php [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+use App\Mail\ShareWithAFriendMail;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Mail;
+
+class ShareWithAFriendService
+{
+    public function inviteFriend(array $data)
+    {
+        $user = Auth::user();
```
Comment (omadoyeabraham) [inline app/Services/ShareWithAFriendService.php]: @enwemasor1995 Do not call Auth::User() in the service. Get the user from the GraphQL resolver. Check the mutation for `WithdrawFromContributionPlanToWallet` to see how to inject the user into the args for the graphql class, you can then access it as $args['user'] and pass that to the service

File: graphql/mutations/share-with-a-friend-mutations.graphql [hunk 6]
```diff
@@ -0,0 +1,4 @@
+extend type Mutation  @middleware(checks: ["auth:api"]) {
+    """ Send an invitation email """
+    ShareWithAFriend(name: String! phone_number: String email: String!): String @field(resolver: "ShareWithAFriend@resolve")
```
Comment (omadoyeabraham) [inline graphql/mutations/share-with-a-friend-mutations.graphql]: @enwemasor1995 Remove the @field resolver, because you have changed resolve to __invoke above

File: resources/views/vendor/mail/html/button.blade.php [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<table class="action" align="center" width="100%" cellpadding="0" cellspacing="0" role="presentation">
```
Comment (omadoyeabraham) [inline resources/views/vendor/mail/html/button.blade.php]: @enwemasor1995 Why did you add all these vendor email files?

**Review by omadoyeabraham — COMMENTED** [review]
@enwemasor1995 Please address my comments

**Comment by eenwemasor** [comment]
@omadoyeabraham  corrections made

**Comment by eenwemasor** [comment]
@omadoyeabraham
Can't find the mutations created with pr
